bench/workgen/workgen.swig

```diff
@@ -142,6 +145,67 @@ WorkgenFrozenClass(TableOptions)
 WorkgenFrozenClass(ThreadOptions)
 WorkgenFrozenClass(WorkloadOptions)
 
+/*
+ * It's a bit tricky to redefine a constructor, but it can be done.
+ * This question had the needed clues to make it work:
+ *   https://stackoverflow.com/questions/33564645/how-to-add-an-alternative-constructor-to-the-target-language-specifically-pytho
+ * It does unfortunately rely on some SWIG implementation details.
+ */
+%inline %{
+    workgen::Context* _new_Context() {
+        return new workgen::Context();
+    }
+%}
+
+%extend workgen::Context {
+%pythoncode %{
+    def __init__(self, parser = None):
```

Comment:
@ddanderson 
I want to try to understand this change a bit better.
Before when we create a `Context` in Python, we were calling into our C++ code in `workgen.cxx`. Now when we create a `Context`, we're calling this thin Python wrapper around the original C++ code?

So it's almost as if we did something like:
```
class Context : public workgen::Context
```
in the C++ and just tacked on some extra methods.
Is that correct?

Reply:
@tetsuo-cpp, yes that's right.  SWIG needs to build an interface for every function, method, constructor, and data object that is exposed in the interface (all the public stuff in `workgen.h`).  SWIG builds up a Python version of every class in the .h, each having (Python) methods that do the magic to jump into the C code (via Python-ready C functions generated in `workgen_wrap.c`.  The `%ignore` told SWIG to pretend that the `Context::Context()` constructor doesn't exist, so SWIG didn't generate the corresponding `Context.__init__`.  This `%pythoncode` defines that __init__ function, does what we want, and calls the `_new_Context()` function to call the real C++ constructor and (gulp) stashes the raw C pointer into `self.this` where SWIG would have put it.  I wish it wasn't this obscure.

Reply:
Thanks for the explanation, that makes sense.
I think those latest changes are better since we're no longer stashing the `this` pointer.